os_migrate/plugins/module_utils/resource.py

```diff
@@ -0,0 +1,236 @@
+        'mtu',
+        'name',
+        'provider_network_type',
+        'provider_physical_network',
+        'provider_segmentation_id',
+        'segments',
+    ]
+    params_from_refs = [
+        'qos_policy_name',
+    ]
+    info_from_sdk = [
+        'availability_zones',
+        'created_at',
+        'id',
+        'project_id',
+        'qos_policy_id',
+        'revision_number',
+        'status',
+        'updated_at',
+    ]
+
+    sdk_params_from_params = [
+        'description',
+        'dns_domain',
+        'is_admin_state_up',
+        'is_default',
+        'is_port_security_enabled',
+        'is_router_external',
+        'is_shared',
+        'is_vlan_transparent',
+        'mtu',
+        'name',
+        'provider_network_type',
+        'provider_physical_network',
+        'provider_segmentation_id',
+        'segments',
+    ]
+    sdk_params_from_refs = [
+        'qos_policy_id',
+    ]
+
+
+    @classmethod
+    def from_sdk(cls, sdk_resource, conn):
+        # >>> Internal composition over inheritance! <<<
+        # This is more future-proof, composable, less restrictive
+        # approach than calling the same method via super.
+        obj = cls()
+        obj._data_from_sdk_and_refs(sdk_resource, _refs_from_sdk(sdk_resource, conn))
+        # Let's not make special lists for sorted params/info, just sort them?
+        obj._sort_param('availability_zone_hints')
+        obj._sort_info('subnet_ids')
+        return obj
+
+    ## TBD: SOME MORE DRY HERE? Can we refactor something more into reusable bits?
+    def import_or_update(self, conn):
+        # Again composition over inheritance.
+        refs = self._refs_from_ser(conn)
+        sdk_params = self._to_sdk_params(net_refs, conn)
+        existing = conn.network.find_network(sdk_params['name'])
+        if existing:
+            if self.needs_update(Network.from_sdk(existing, conn)):
+                conn.network.update_network(sdk_params['name'], **sdk_params)
+                return True
+        else:
+            conn.network.create_network(**sdk_params)
+            return True
+        return False  # no change done
+    ### END TBD
+
+    def _to_sdk_params(self, refs):
+        # Same here re composition over inheritance. We don't make
+        # much use of it here but more complex resources probably
+        # will.
+        sdk_params = {}
+        self._sdk_params_from_params_and_refs(sdk_params, refs)
+        return sdk_params
+
+    def _refs_from_sdk(sdk_resource, conn):
+        # would be just the same as what we already have in network.py
+
+    def _refs_from_ser(self, conn):
+        # would be just the same as what we already have in network.py
+
+
+
+
+# EXAMPLE in import_network module which gets 'data' as a parameter:
+net = Network.from_data(data)
+# there would be more if/else around this, checking existing network
+# perhaps but you get the idea
+conn.network.create_network(net.sdk_params(conn))
```

Comment:
This was what i thought of initially but we can probably make the interface nicer with a full encapsulation of `import_or_update` as written below.